### ISSUE TITLE
Feature request: match incomplete string at beginning of line

### ISSUE THREAD

**Issue by kergoth** [Tarrasch/zsh-bd#11]
It seems that https://github.com/vigneshwaranr/bd supports this: "You can simply type bd <starting few letters> like bd s or bd si", whereas with this, it seems we have to tab-complete.

**Comment by Tarrasch** [comment]
Yes. I think it's a bad feature and I intentionally didn't put it in zsh-bd. `bd` is supposed to be used interactively and getting the confirmation from the computer that "yes, you spelled the suffix correctly" through tab completion is a good thing. I figured you always want this.

Please note that bash-bd got autocompletion after they got the partial input feature.
